ext/sockets/sockets.c

```diff
@@ -1898,6 +1907,7 @@ PHP_FUNCTION(socket_set_option)
 		case SO_SNDTIMEO: {
 			const char sec_key[] = "sec";
 			const char usec_key[] = "usec";
+			bool failed;
 
 			convert_to_array(arg4);
```

Comment:
This could also be updated to check that the value is an array, as if it is not the `zend_hash_str_find` calls below will fail. Similarly for the other function.

Reply:
This `convert_to_array()` appears to be superfluous.